src/inc/msquic.h

```diff
@@ -890,6 +890,7 @@ typedef struct QUIC_SCHANNEL_CREDENTIAL_ATTRIBUTE_W {
 #define QUIC_PARAM_CONN_STATISTICS_V2                   0x05000016  // QUIC_STATISTICS_V2
 #define QUIC_PARAM_CONN_STATISTICS_V2_PLAT              0x05000017  // QUIC_STATISTICS_V2
 #define QUIC_PARAM_CONN_ORIG_DEST_CID                   0x05000018  // uint8_t[]
+#define QUIC_PARAM_CONN_RELIABLE_RESET_ENABLED          0x05000019  // uint8_t (BOOLEAN)
```

Comment:
I was thinking about this more... and I think putting this in the QUIC_SETTINGs is a better route. You already added it to the internal settings, but add it to the external one (at the end so you don't break back compat), and update settings.c accordingly. So then you don't have this.

You will also need to add a new Connection event to indicate if the feature was negotiated; and it's only delivered for apps that try to enable it.

I'd be happy to talk more of the details next week.

Reply:
Good point, that makes a lot of sense. I added `ReliableResetEnabled` in QUIC_SETTINGS in msquic.h. 

In terms of scope for this PR, I think including all the code for the variable initializations and the logic for the setup and teardown could be a good stopping point. I'd love to discuss this more with you more if you have the time!